tests/trilinos/trilinos_sparsity_pattern_03.cc

```diff
@@ -39,7 +39,7 @@ void test ()
   columns.add_range(3*myid, 3*myid+3);
   columns.compress();
 
-  sp.reinit(rows, columns, MPI_COMM_WORLD, 0);
+  sp.reinit(rows, columns, MPI_COMM_WORLD, (dealii::types::global_dof_index)0);
```

Comment:
If you hand in a dynamic sparsity pattern, you can skip the Trilinos sparsity pattern altogether and directly fill the dynamic sparsity pattern to the Trilinos sparse matrix. We needed the TrilinosSparsityPattern back in 2008/2009 when it was written because our sparsity pattern couldn't handle the data exchange necessary to create all matrix entries back then. But now we do have that functionality in DynamicSparsityPattern (@tjhei wrote that in late 2009 IIRC) so if we insist in changing things, let's change enough. :-)
On the other hand, a Trilinos sparsity pattern can be filled completely analogous to our own sparsity pattern, so if we have a Trilinos sparsity pattern, the default way to work with it is to not fill it from another sparsity pattern but rather fill it with make_sparsity_pattern and friends.